lib/rucio/gateway/account.py

```diff
@@ -59,8 +59,9 @@ def add_account(
     validate_schema(name='account', obj=account, vo=vo)
 
     kwargs = {'account': account, 'type': type_}
-    if not rucio.gateway.permission.has_permission(issuer=issuer, vo=vo, action='add_account', kwargs=kwargs, session=session):
-        raise rucio.common.exception.AccessDenied('Account %s can not add account' % (issuer))
+    auth_result = rucio.gateway.permission.has_permission(issuer=issuer, vo=vo, action='add_account', kwargs=kwargs, session=session)
+    if not auth_result.allowed:
```

Comment:
Maybe I don't completely understand how you've set this up, but shouldn't you be able to just do `if not auth_result`? `PermissionResult.__bool__` already returns allowed.

Reply:
Yes, that would also work. I thought this might make it clearer that the type is a `PermissionResult` rather than just a bool but I don't have strong feelings either way.